ARCHITECTURE.md

```diff
@@ -0,0 +1,76 @@
+# Architecture of SHARE/Trove
+
+This document is a starting point and reference to familiarize yourself with this codebase.
+
+## Bird's eye view
+In short, SHARE/Trove takes metadata records (in any supported input format),
+ingests them, and makes them available in any supported output format
+
+## Code map
+
+A brief look at important areas of code as they happen to exist now.
+
+### Static configuration
+
+`share/schema/` describes the "normalized" metadata schema/format that all
+metadata records are converted into when ingested.
+
+`share/sources/` describes a starting set of metadata sources that the system
+could harvest metadata from -- these will be put in the database and can be
+updated or added to over time.
+
+`project/settings.py` describes system-level settings which can be set by
+environment variables (and their default values), as well as settings
+which cannot.
+
+`share/models/` describes the data layer using the [Django](https://www.djangoproject.com/) ORM.
+
+### Harvest and ingest
+
+`share/harvest/` and `share/harvesters/` describe how metadata records
+are pulled from other metadata repositories.
+
+`share/transform/` and `share/transformers/` describe how raw data (possibly
+in any format) are transformed to the "normalized" schema.
+
+`share/regulate/` describes rules which are applied to every normalized datum,
+regardless where or what format it originally come from.
+
+`share/metadata_formats/` describes how a normalized datum can be formatted
+into any supported output format.
+
+`share/tasks/` runs the pipeline: harvest -> transform -> regulate -> format
+
+### Outward-facing views
+
+`share/search/` describes how the search indexes are structured, managed, and
+updated when new metadata records are introduced -- this provides a view for
+discovering items based on whatever search criteria.
+
+`share/oaipmh/` describes the [OAI-PMH](https://www.openarchives.org/OAI/openarchivesprotocol.html)
+view for harvesting metadata from SHARE/Trove in bulk.
+
+`api/` describes a mostly REST-ful API that's useful for inspecting records for
+a specific item of interest.
+
+### Testing
+
+`tests/` are tests.
+
```

Comment:
Things that are undocumented but maybe should be:

`share/admin/` => "Django-app for administrative access to SHARE pipeline / db"

`share/ingest/` => ???

`share/janitor/` => ???

`osf_oauth2_adapter/` =>  Django app to support logging in to SHARE via OSF

`bots/elasticsearch/` => ???

`db/` => default Pg database configurations (I think?)

`project/` => Django app that dispatches to other django apps in the repo (I think?)

`Vagrantfile` => Is this deletable?

`bin/share`, `bootstrap.sh`, `up.sh`, `share/bin/*` => files behind sharectl and initial setup.  Some (bootstrap.sh) may be deletable?  Maybe this can be put off until share cmd consolidation.